DevoxxClientMobile/src/main/java/com/devoxx/util/DevoxxNotifications.java

```diff
@@ -121,6 +138,82 @@ public final void removeFavoriteSessionNotifications(Session session) {
+            if (service.retrieveSessions().size() > 0) {
+                addAlreadyRatingNotifications();
+                service.retrieveSessions().removeListener(ratingListener);
+            }
+        };
+        service.retrieveSessions().addListener(ratingListener);
+    }
+
+    /**
+     * Called after the application has started and pre-loading ends.
+     * At this point, we have all the notifications available, and we can remove
+     * the listener (so new notifications are not treated as already scheduled) and
+     * send them to the Local Notifications service at once
+     */
+    public void preloadingRatingNotificationsDone() {
+        if (ratingListener != null) {
+            service.retrieveSessions().removeListener(ratingListener);
+            ratingListener = null;
+
+            if (! dummyNotificationMap.isEmpty()) {
+
+                // 1. Add all dummy notifications to the notification map at once.
+                // These need to be present all the time (as notifications will
+                // be opened always some time after they were delivered).
+                // Adding these dummy notifications doesn't schedule them on the
+                // device and it doesn't cause duplicate exceptions
+                notificationsService.ifPresent(ns ->
+                        ns.getNotifications().addAll(dummyNotificationMap.values()));
+            }
+
+            // process notifications at once
+            List<Notification> notificationList = new ArrayList<>(ratingNotificationMap.values());
+
+            if (! notificationList.isEmpty()) {
+
+                // 2. Schedule only real future notifications
+                final ZonedDateTime now = ZonedDateTime.now(service.getConference().getConferenceZoneId());
+                notificationsService.ifPresent(ns -> {
+                    for (Notification n : notificationList) {
+                        if (n.getDateTime() != null && n.getDateTime().isAfter(now)) {
+                            // Remove notification before scheduling it again,
+                            // to avoid duplicate exception
+                            final Notification dummyN = dummyNotificationMap.get(n.getId());
+                            if (dummyN != null) {
+                                if (LOGGING_ENABLED) {
+                                    LOG.log(Level.INFO, String.format("Removing notification %s", n.getId()));
+                                }
+                                ns.getNotifications().remove(dummyN);
+                            }
+                            if (LOGGING_ENABLED) {
+                                LOG.log(Level.INFO, String.format("Adding rating notification %s", n.getId()));
+                            }
+                            ns.getNotifications().add(n);
+                        }
+                    }
+                });
+            }
+
+            dummyNotificationMap.clear();
```

Comment:
Also, if both type of notifications use the same `dummyNotificationMap`, clearing it when one process finished first could remove the notifications from the other process